### DISCUSSION TITLE
1.1

### DISCUSSION THREAD

**Discussion by joethephish** [inkle/ink#800] [General]
# Version 1.1

First, a little tip: We strongly suggest never updating to a major new version of ink for a game that's in production! While we aim for backward compatibility, we don't absolutely guarantee stability between versions.

Now, on to the fun part:

## New feature: Dynamic tags, and tags in choices

*A big thanks to [BetterUp](https://www.betterup.com) for sponsoring this feature.*

You can now write ink rather than just plain text within tags:

```
VAR subject = "dog"
This is a picture of a {subject}. # image_of_{subject}.jpg
```

You can also directly use tags within choices, in all the places you can use text within choices. Here's a simple example:

```
+ [Take the apple # hand_holding_apple.jpg]
```

And here's an example that uses tags in every spot within the choice:

```
+ Start of choice text # tag both [Choice only text # choice only tag] This is after the choice is taken # post choice tag
```

On the game code side, the choices themselves now have a list of choices available on them too:

```
public class Choice : Runtime.Object {
    ...
    public List<string> tags;
    ...
}
```

## Other changes

- **BREAKING CHANGE**: Doing `someList ? ()` (i.e. specifically checking for a list containing an empty list) now returns `false` rather than `true`, since this makes more logical sense in most circumstances and reduces the amount of explicit checks for that case.
- Update to .NET 6.0
- Documentation fixes (thanks @deviant-dev, @bitbutter, @tomkail, @owennewburn, @Atlas48 )
- inklecate no longer outputs BOM in output (thanks @Fendse)
- It's now possible to use `->-> some_variable_divert_target`.
- Add some multi-flow utility functions (thanks @russellquinn)
- Various external functions and list API improvements (Thanks @tomkail)
- Improve plugin API (Thanks @elliotherriman)
- Fixes for origin names being wrong on lists

**New ink unity integration plugin is also on its way, hopefully within the next day or two...**

<hr /><em>This discussion was created from the release <a href='https://github.com/inkle/ink/releases/tag/'>1.1</a>.</em>

**Comment by russellquinn** [comment]
Woo!

**Comment by chamik** [comment]
great work! :)

**Comment by maddawik** [comment]
Cheers and thanks for more features around tags!

Also noticed a minor typo:

> On the game code side, the choices themselves now have a list of **_tags_** available on them too:
